tests/test_exposure_table.py

```diff
@@ -0,0 +1,225 @@
+    np.testing.assert_allclose(shift[2], (idx + 2), rtol=0, atol=1e-15)
+
+    # Read the full structured array with units and convert to cm
+    shift, safe = ExposureData({"field": "shift", "to_unit": "cm"}, config, list)
+    assert safe == True
+    assert len(shift) == 3
+    np.testing.assert_allclose(shift[0], idx * 100, rtol=0, atol=1e-15)
+    np.testing.assert_allclose(shift[1], (idx + 1) * 100, rtol=0, atol=1e-15)
+    np.testing.assert_allclose(shift[2], (idx + 2) * 100, rtol=0, atol=1e-15)
+
+
+def test_exposure_table():
+    table = QTable()
+    # For testing single and multi indexing
+    table["idx"] = [0, 1, 2, 3]
+    table["idx0"] = [0, 0, 1, 1]
+    table["idx1"] = [0, 1, 0, 1]
+    table["unitless"] = np.array(table["idx"], dtype=float)
+    table["angle1"] = table["idx"] * u.arcsec
+    table["angle2"] = table["angle1"].to(u.deg)
+    # structured array column
+    tilt_dtype = np.dtype([("rx", "<f8"), ("ry", "<f8")])
+    table["tilt"] = np.array([(0, 1), (1, 2), (2, 3), (3, 4)], dtype=tilt_dtype)
+    table["tilt"].unit = u.arcsec
+    # Add columns with other types
+    table["int"] = np.array(table["idx"], dtype=int)
+    table["bool"] = np.array([True, False, True, False], dtype=bool)
+    table["str"] = ["a", "b", "c", "d"]
+    # And some other units
+    table["length"] = table["idx"] * u.m
+    # Structured array with length units
+    shift_dtype = np.dtype([("dx", "<f8"), ("dy", "<f8"), ("dz", "<f8")])
+    table["shift"] = np.array(
+        [(0, 1, 2), (1, 2, 3), (2, 3, 4), (3, 4, 5)], dtype=shift_dtype
+    )
+    table["shift"].unit = u.m
+
+    table.pprint_all()
+
+    with TemporaryDirectory() as tmpdir:
+        for ext in [".fits", ".ecsv", ".parq"]:
+            file_name = tmpdir + "/exposure_table" + ext
+            table.write(file_name, overwrite=True)
+
+            config = {
+                "input": {
+                    "exposure_table": {
+                        "file_name": file_name,
+                        "keys": ["idx"],
+                        "values": [0],
+                    },
+                },
+            }
+
+            # Check single indexing during table load
+            for idx in range(4):
+                config["input"]["exposure_table"]["values"] = [idx]
+                galsim.config.RemoveCurrent(config["input"]["exposure_table"])
+                galsim.config.ProcessInput(config)
+                check_exp_data(config, idx)
+
+            # Check multi indexing during table load
+            for idx0 in range(2):
+                for idx1 in range(2):
+                    config["input"]["exposure_table"]["keys"] = ["idx0", "idx1"]
+                    config["input"]["exposure_table"]["values"] = [idx0, idx1]
+                    galsim.config.RemoveCurrent(config["input"]["exposure_table"])
+                    galsim.config.ProcessInput(config)
+                    idx = idx0 * 2 + idx1
+                    check_exp_data(config, idx)
```

Comment:
Rather than just checking explicit calls to ExposureData, it's probably also worth adding some values in the config itself that use the input type.  Things that use `{ 'type': 'ExposureData', 'field': ... }`, since that's the syntax that users will actually use.

Reply:
Sure.  I added a test that actually creates a batoid telescope, similar to what I'm doing for OR4.